src/main/kotlin/org/rust/ide/annotator/RsErrorAnnotator.kt

```diff
@@ -163,16 +164,16 @@ class RsErrorAnnotator : RsAnnotatorBase(), HighlightRangeExtension {
     }
 
     private fun checkReferenceIsPublic(ref: RsReferenceElement, o: PsiElement, holder: AnnotationHolder) {
-        val element = ref.reference.resolve() as? RsVisible ?: return
+        val element = ref.reference.resolve() as? RsVisibilityOwner ?: return
         val oMod = o.contextStrict<RsMod>() ?: return
         if (element.isVisibleFrom(oMod)) return
-
+        val withinOneCrate = element.crateRoot == (o as RsElement).crateRoot
         val error = when {
             element is RsNamedFieldDecl -> {
                 val structName = element.ancestorStrict<RsStructItem>()?.crateRelativePath?.removePrefix("::") ?: ""
-                RsDiagnostic.StructFieldAccessError(ref, ref.referenceName, structName)
+                RsDiagnostic.StructFieldAccessError(ref, ref.referenceName, structName, MakePublicFix.createIfCompatible(element, withinOneCrate))
```

Comment:
Please, create the fix inside `RsDiagnostic.StructFieldAccessError` inside `prepare` method

Reply:
(I fount that a lot of things already made wrong, and so it's hard and not reasonable to move quickfixes creation into `prepare`...)